ui/pages/settings/flask/snaps-list-tab/view-snap/view-snap.container.js

```diff
@@ -0,0 +1,35 @@
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { getSubjectMetadata } from '../../../../../selectors';
+import { removePermittedAccount } from '../../../../../store/actions';
+import ViewSnap from './view-snap.component';
+
+const mapStateToProps = (state, ownProps) => {
+  const { snap } = ownProps;
+  const subjectMetadata = getSubjectMetadata(state);
+  const connectedSubjects = [];
+  Object.entries(subjectMetadata).forEach(([_, entry]) => {
+    if (entry.subjectType === 'snap') {
+      const { iconUrl, name, origin } = entry;
+      connectedSubjects.push({ iconUrl, name, origin });
+    }
+  });
+  return {
+    snap,
+    connectedSubjects,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    onDisconnect: (origin, address) => {
+      dispatch(removePermittedAccount(origin, address));
+    },
+  };
+};
+
+export default compose(
+  withRouter,
+  connect(mapStateToProps, mapDispatchToProps),
+)(ViewSnap);
```

Comment:
I think we underlying functional components we avoid the `container` pattern and use methods like `useSelector` and `useDispatch`, but given our rush here, I think this is fine for now.